server/Entities/UserRepository.cs

```diff
@@ -22,5 +22,13 @@ public void CreateUser(User user)
             RepositoryContext.Users.Add(user);
             RepositoryContext.SaveChanges();
         }
+
+        public void UpdateUserPassword(User user)
+        {
+            var newPassword = user.Password;
```

Comment:
no checks here? just arbitrary user can update password for anyone

Reply:
I put the checks on the person who calls the method
but Yes, it would be correct to add a check there as well